apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
```

Comment:
note: if matej likes my proposal, we'll need to reverse the meaning here; so replace "enabled" by "disabled" and resolve the promise with "!value".

Also, nit: please put the name in a constant, like `READ_AHEAD_THREADS_KEY`